### PR TITLE
Reworks bolas.

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
You now must "spin up" a bola to throw it.
activating throw intent with a bola in hand will start the "spin"
each time the progress bar fills you get a percentage of your max range and throw speed added on to the bola. 
switching hands stops the spin.
as of right now, all bolas need 3 spins (3 seconds of time) to get the max range
you can move while you spin the bola.

Decreases the breakout time of energy bolas by 2 seconds (to 4 seconds) [up for debate, not sure about this change]
makes the ebola catchable /blockable by shields.
and the ebola now properly plays its hit sound!

## Why It's Good For The Game
Bolas are reaaally good right now.
Security can throw a bola at someone within a moments notice, with zero time to react, then they are easy picking for batoning/disabling. with practically no counterplay. In the time it takes to remove a bola you can get stam critted twice over, so your only chance is to dispatch security quickly with a ranged weapon, which you cannot do as easily anymore.

What this does it, it gives the antag time to notice that sec has a bola, spot it moving, toggle on throw intent to catch. giving the item counterplay. This does however nerf cult and reinforced bolas. might need tweaking.

## Images of changes
[dreamseeker_kICvpThrgH.webm](https://user-images.githubusercontent.com/69320440/179061028-a594fafe-1a7d-4a54-9017-7f54e201ed0c.webm)

[dreamseeker_OxRX1TDZ66.webm](https://user-images.githubusercontent.com/69320440/179061065-6eaf1952-5aeb-4bee-8c7a-955e22b01283.webm)

[dreamseeker_K4gOgLSLvZ.webm](https://user-images.githubusercontent.com/69320440/179061130-7991cda0-7db0-4ba0-8a92-80763c90917a.webm)

## Changelog
:cl: Hal 9000
tweak: bolas now take time to "spin up" to throw, the longer you spin them, the further you can throw them.
tweak: energy bolas can be blocked/caught
tweak: energy bolas take 4 seconds to remove instead of 6 seconds.
/:cl:

### CODE REVIEW

**Comment by Inadvizable** [comment]
This looks cool. I am excited to see how it performs.

**Comment by KittyNoodle** [comment]
does this effect cult bolas, if so please make them take less time to spin up

**Comment by roarbark** [comment]
cult bolas are already really really strong. this change might not affect cult balance that negatively. the big difference between cult and reinforced bolas is if you get a cult bola landed on you, you get round removed, but if you get a reinforced bola on you, you only get killed.

also, thoughts on making the wind-ups get lost after like, 5 or 10 seconds? one situation after this change as is might be seccies walking through maints with their bola already wound up, throwing by reaction- which is the next best thing to what it used to be. also, once a bola is wound up, do those stacks persist if i switch my hands? maybe they shouldn't

**Comment by KittyNoodle** [comment]
> cult bolas are already really really strong. this change might not affect cult balance that negatively. the big difference between cult and reinforced bolas is if you get a cult bola landed on you, you get round removed, but if you get a reinforced bola on you, you only get killed.
> 
> also, thoughts on making the wind-ups get lost after like, 5 or 10 seconds? one situation after this change as is might be seccies walking through maints with their bola already wound up, throwing by reaction- which is the next best thing to what it used to be. also, once a bola is wound up, do those stacks persist if i switch my hands? maybe they shouldn't

if cult is round removing people as a non target without even attempting to convert pre-halo ahelp that

**Comment by roarbark** [comment]
> > cult bolas are already really really strong. this change might not affect cult balance that negatively. the big difference between cult and reinforced bolas is if you get a cult bola landed on you, you get round removed, but if you get a reinforced bola on you, you only get killed.
> > also, thoughts on making the wind-ups get lost after like, 5 or 10 seconds? one situation after this change as is might be seccies walking through maints with their bola already wound up, throwing by reaction- which is the next best thing to what it used to be. also, once a bola is wound up, do those stacks persist if i switch my hands? maybe they shouldn't
> 
> if cult is round removing people as a non target without even attempting to convert pre-halo ahelp that, more often round removal comes from the reinforced bolas

i was referring to cult vs sec, because like, it wont matter to converting because you can just like, use stunhand out of disguise

**Comment by hal9000PR** [comment]
> persist if i switch my hands? maybe they shouldn't

It doesn't.

> if cult is round removing people as a non target

Security exists

> also, thoughts on making the wind-ups get lost after like, 5 or 10 seconds

I dunno, atleast you can catch them now with the super obvious sprite.

**Comment by Freakycrafter** [comment]
Found a bug: if you toggle throwmode on with a bola, then let it charge and put it in your backpack it stays charged, essentially letting you keep it charged indefinitely. It will return to normal after thrown though (as far as i can tell).

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -617,6 +617,9 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 		throw_mode_off()
 	else
 		throw_mode_on()
+	var/obj/item/I = get_active_hand()
+	if(I)
```
Comment (farie82) [inline code/modules/mob/living/carbon/carbon.dm:621]: What happens if it ain't in your active hand? It prob won't start spinning right?

Comment (hal9000PR) [reply code/modules/mob/living/carbon/carbon.dm:621]: yep. thats intended

**Review by farie82 — APPROVED** [review]

**Pull request merged**